File: psets/pset1/marioless/mario.c
/*
    This program outputs a mario pyramid of desired size.
    Jeet Khamar, CS50 - 26/05/2022.
*/

#include <cs50.h>
#include <stdio.h>

int main(void)
{
    int height;

    //Verifying whether the user input is within 1 and 8.
    do
    {
        height = get_int("Height: ");
    }
    while (height < 1 || height > 8);

    //This loop iterates for every row in the pyramid.
    for (int i = 1; i <= height; i++)
    {
        //This loop prints the number of spaces in a particular line.
        for (int space = height - i; space > 0; space--)
        {
            printf(" ");
        }

        //This loop prints the number of hashes in a particular line.
        for (int hash = 1; hash <= i; hash++)
        {
            printf("#");
        }

        //This moves the cursor to the next line after the row printing is complete.
        printf("\n");
    }
}

File: psets/pset6/sentimental-mario-more/mario.py
# This program prints a mario pyramid of user desired size.

from cs50 import get_int

# This loop checks whether the user input is between 1 and 8.
while True:
    height = get_int("Height: ")
    if height > 0 and height < 9:
        break

for i in range(1, height + 1):
    print(" " * (height - i), end="")
    print("#" * i, end="")
    print("  ", end="")
    print("#" * i)